app/employee-app/backend/service.bal

```diff
@@ -0,0 +1,752 @@
+        if employees is error {
+            string customError = "Error getting employee basic information for employees";
+            log:printError(customError, employees);
+            return <http:InternalServerError>{
+                body: {
+                    message: customError
+                }
+            };
+        }
+        return employees;
+    }
+
+    # Organization structure data retrieval.
+    #
+    # + return - Internal Server Error or Organization structure data
+    resource function get org\-structure() returns types:OrgStructure|error {
+
+        if orgStructureCache.hasKey(types:ORG_STRUCTURE_CACHE_KEY) {
+            return <types:OrgStructure>check orgStructureCache.get(types:ORG_STRUCTURE_CACHE_KEY);
+        }
+
+        types:OrgStructure orgStructure = {
+            businessUnits: check database:getOrgDetails(filter = {}, 'limit = types:DEFAULT_LIMIT, offset = 0),
+            companies: check database:getCompanies(filter = {}, 'limit = types:DEFAULT_LIMIT, offset = 0),
+            careerPaths: check database:getCareerFunctions(filter = {}, 'limit = types:DEFAULT_LIMIT, offset = 0),
+            employmentTypes: [types:PERMANENT, types:INTERNSHIP, types:CONSULTANCY]
+        };
+        check orgStructureCache.put(types:ORG_STRUCTURE_CACHE_KEY, orgStructure);
+        return orgStructure;
+    }
+
+    # Retrieve specific recruit data from the database.
+    #
+    # + return - recruit data or error response
+    resource function get recruits/[int recruitId](http:RequestContext ctx)
+        returns types:Recruit|http:Forbidden|error => check database:getRecruit(recruitId);
+
+    # Retrieve all recruits data from the database.
+    #
+    # + statusArray - Array of recruit statuses to filter the data  
+    # + 'limit - Limit of the data  
+    # + offset - Offset of the data
+    # + return - All Selected recruits data or error response
+    resource function get recruits(database:RecruitStatus[]? statusArray, int? 'limit, int? offset)
+        returns types:Recruit[]|error => check database:getRecruits(statusArray, 'limit, offset);
+
+    # Retrieve compensations for a specific recruit's employment type and the company location.
+    #
+    # + recruitId - ID of the recruit  
+    # + return - Compensation or error response
+    resource function get recruits/[int recruitId]/compensation(http:RequestContext ctx)
+        returns types:Compensation[]|http:Forbidden|error {
+
+        authorization:CustomJwtPayload userInfo = check ctx.getWithType(authorization:HEADER_USER_INFO);
+        if !authorization:checkRoles([authorization:authorizedRoles.recruitmentTeamRole],
+                userInfo.groups) {
+            return <http:Forbidden>{body: {message: "Insufficient privileges!"}};
+        }
+
+        types:Recruit {employmentType, companyLocation} = check database:getRecruit(recruitId);
+        companyLocation = regex:replaceAll(companyLocation, " ", "");
```

Comment:
The use of ballerina/regex is deprecated. Instead of this use [regexp](https://central.ballerina.io/ballerina/lang.regexp/latest) in ballerina

Reply:
ACK